### ISSUE TITLE
iOS cannot discover a host running on Windows

### ISSUE THREAD

**Issue by gomaze1031** [Skyost/Bonsoir#83]
**Describe the bug**
iOS cannot discover a host running on Windows.

I tested as follows:
1. Connect all devices to the same WiFi (and checked they are assigned to 192.168.0.x)
2. Start hosting on Windows and macOS
3. iOS discovers a host running on macOS
4. iOS cannot discover a host running on Windows
5. Android discovers a host running on macOS and Windows

Version info:
- Bonsoir 5.1.8
- Windows 10 22H2(19045.3930)
- macOS 14.2.1
- iOS 17.2.1
- Android 14

FYI, there is no errors or other logs printed. Not sure it is a bug.
Thanks for your nice work!

**Comment by Skyost** [comment]
Are you sure you're discovering the correct type ? Do you have a snippet or something ?

**Comment by gomaze1031** [comment]
I think I discovered the correct type.
Here are some snippets.

startDiscovery()
```
_dnsHost = BonsoirBroadcast(
  service: BonsoirService(
    name: "my_bonsoir_server",
    type: "_dns-service._tcp",
    port: 56000,
  ),
);

await _dnsHost?.ready;
await _dnsHost?.start();
```

startScan()
```
BonsoirDiscovery discovery = BonsoirDiscovery(type: "_dns-service._tcp");
_mDnsDiscovery.discovery = discovery;
await discovery.ready;

discovery.eventStream!.listen((event) async {
  if (event.type == BonsoirDiscoveryEventType.discoveryServiceFound) {
    debugPrint('Service found : ${event.service?.toJson()}');

    if (event.service is ResolvedBonsoirService) {
      ...
      return;
    }

    // Should be called when the user wants to connect to this service.
    await event.service!.resolve(discovery.serviceResolver);
  } else if (event.type == BonsoirDiscoveryEventType.discoveryServiceResolved) {
    debugPrint('Service resolved : ${event.service?.toJson()}');
    ...
  } else if (event.type == BonsoirDiscoveryEventType.discoveryServiceLost) {
    debugPrint('Service lost : ${event.service?.toJson()}');
    ...
  }
});

await discovery.start();
```

Not sure it helps you but this is all I've got.

* Side question: when the event.type is "discoveryServiceFound", event.service can be "ResolvedBonsoirService"?

**Comment by Skyost** [comment]
> Side question: when the event.type is "discoveryServiceFound", event.service can be "ResolvedBonsoirService"?

Nop !

**Comment by jaween** [comment]
I am running into this same issue:
 * Windows, Mac and Linux apps are using Bonsoir to broadcast, Android and iPhone apps are using Bonsoir to discover
 * All devices are on the same Wi-Fi network running the same version of Bonsoir
 * Android sees all three broadcasts, iPhone sees two, missing Windows
 * No Windows discovery events show up on the iPhone
 * Both phones successfully resolve the hosts they see


Versions:
 * Bonsoir 5.1.8
 * Windows 11 Pro 23H2 (22631.3155)
 * macOS 14.3.1
 * Linux Ubuntu 22.04.3 LTS (Avahi Daemon 0.8)
 * iOS 15.7.9
 * Android 13

**Comment by Skyost** [comment]
Hey,

This could be the same problem than https://stackoverflow.com/questions/60925721/my-ios-macos-devices-are-unable-to-resolve-my-bonjour-zeroconf-service-created-o. Does it work with a `ResolvedBonsoirService` ?

**Comment by jaween** [comment]
(Sorry maybe I'm misunderstanding)

With the link it looks like they used an npm package to broadcast a bonjour service which iOS discovered sucessfully, but couldn't resolve it without an extra parameter in the broadcast. But in our case iOS is not even discovering the service. 

`ResolvedBonsoirService` works with the discovered Mac and Linux services, and I assume if the Windows service was discovered, it would work too.

**Comment by Skyost** [comment]
You're right, I've misread the SO post. I'll continue to investigate on my side, but feel free to post here if you got something new.

**Comment by Styx-S** [comment]
same problem, but I found that if I add the "attributes" parameter (like demo), it can be successful resolved
there must be some differences in win native code 

badcase
```
// flutter
BonsoirService service = BonsoirService(
      name: name, 
      type:type
      port: port,
);

//  iOS get resolved Error with
{
NSNetServicesErrorCode = "-72007";
NSNetServicesErrorDomain = 10;
}
```

he demo is written like this:
```
BonsoirService service = BonsoirService(
      name: name, 
      type:type
      port: port,
     //fill any of the os or uuid parameters can avoid this problem
      attributes: {'os': 'Windows', 'uuid': const Uuid().v6(config: V6Options(null, null, null, null, selfServiceName!.codeUnits))}
)
```

**Comment by lexxxel** [comment]
Have the same issue, but discovered something strange. When I use an iPhone App as bounjour browser, I can't see the windows host. However, when I start a Linux server on parallel, both server become visible. 
To me, this looks like something is broken with the service topic (or whatever it's called).

**Comment by lexxxel** [comment]
Another finding. Windows does not publish its IP address under host, but its link local URL.

**Comment by lexxxel** [comment]
Found a fix, fill the attributes: 
```dart
BonsoirService service = BonsoirService(
      name: name,
      type: type,
      port: port,
      attributes: {"DUMMY": "iOSFix"}, // <- content does not matter, but you have to set a TXT info.
    );
```

This is so stupid, I didn't believe it first.

**Comment by Skyost** [comment]
@lexxxel Good find ! To summarize, you @lexxxel say we need to publish a TXT record so that Windows hosts can be discovered by Darwin devices ?

**Comment by lexxxel** [comment]
Yes, at least an iOS device using this lib for discovery. When I use avahi browser UI under Linux or an equivalent on macOS, I do NOT see the service. But the service is always discovered by the avahi browser cli tool under Linux (with and without a TXT) or on iOS and Android (with and without a TXT) when using the lib.

**Comment by isaanyoha** [comment]
how do I get the IP address of the discovered device (service)?

**Comment by lexxxel** [comment]
> how do I get the IP address of the discovered device service?
> 
> A code snippet will go a long way. Thank you!

There's a readme.md, you might wanna read. 
You will even find a `code snippet` there. Focus your attention around: `event.service!.resolve(discovery.serviceResolver)`.

**Comment by isaanyoha** [comment]
Yeah, I know about this line ===> event.service!.resolve(discovery.serviceResolver), I read the read.md .

However, I still have not seen the way to get the IP address after the resolving the service. I want to pass the IP address into Socket.connect(ip, port)

**Comment by lexxxel** [comment]
@isaanyoha I do not think your comments are related to the issue here. Please search for support somewhere else. As a suggestion: start a debugger and look into the data structures of said callback.

**Comment by Skyost** [comment]
> I do not think your comments are related to the issue here.

Yep !

@isaanyoha Use `service.resolve()` and check #96.
